Monika After Story/game/script-python.rpy

```diff
@@ -0,0 +1,1205 @@
+
+init 4 python in mas_ptod:
+    # to simplify unlocking, lets use a special function to unlock tips
+    import datetime
+    import store.evhand as evhand
+
+    M_PTOD = "monika_ptod_tip{:0>3d}"
+
+    def has_day_past_tip(tip_num):
+        """
+        Checks if the tip with the given number has already been seen and
+        a day has past since it was unlocked.
+        NOTE: by day, we mean date has changd, not 24 hours
+
+        IN:
+            tip_num - number of the tip to check
+
+        RETURNS:
+            true if the tip has been seen and a day has past since it was
+            unlocked, False otherwise
+        """
+        # as a special thing for devs
+        if renpy.game.persistent._mas_dev_enable_ptods:
+            return True
+
+        tip_ev = evhand.event_database.get(
+            M_PTOD.format(tip_num),
+            None
+        )
+
+        if tip_ev is None:
+            return False
+
+        # otherwise, unlocked date is our key
+        if tip_ev.unlock_date is None or tip_ev.shown_count == 0:
+            return False
+
+        # now check the actual day
+        return (
+            datetime.date.today() - tip_ev.unlock_date.date() 
+            >= datetime.timedelta(days=1)
+        )
+
+    def has_day_past_tips(*tip_nums):
+        """
+        Variant of has_day_past_tip that can check multiple numbers
+
+        SEE has_day_past_tip for more info
+
+        RETURNS:
+            true if all the given tip nums have been see nand a day has past
+                since the latest one was unlocked, False otherwise
+        """
+        for tip_num in tip_nums:
+            if not has_day_past_tip(tip_num):
+                return False
+
+        return True
+
+
+# The initial event is getting Monika to talk about python
+# this must be hidden after it has been completed
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_ptod_tip000",
+            category=["python tips"],
+            prompt="Can you teach me about Python?",
+            pool=True
+        )
+    )
+
+label monika_ptod_tip000:
+    m 3eub "You want to learn about Python?"
+    m 3hub "I'm glad!"
+    m 1lksdlb "I don't know {i}that{/i} much about programming, but I will try my best!"
+    m 1esa "Let's start with what Python even is."
+
+    $ hideEventLabel("monika_ptod_tip000", depool=True)
+    jump monika_ptod_tip001
+
+###############################################################################
+init 5 python:
+    addEvent(
+        Event(
+            persistent.event_database,
+            eventlabel="monika_ptod_tip001",
+            category=["python tips"],
+            prompt="What is Python?"
+        )
+    )
+
+label monika_ptod_tip001:
+    # first enable this event
+    $ tip_ev = mas_getEV("monika_ptod_tip001")
+    $ tip_ev.pool = True
+    $ tip_ev.unlocked = True
+
+    m 1esa "Python was created by Guido Van Rossum in the early 90s."
```

Comment:
'90s.

Reply:
Also, if we're going into the history, I think it's be remiss not to mention what Python is named for (Monty Python) and maybe one other fun tidbit before getting to nitty gritty. but again #nitpick

Reply:
I actually dont want too much history here. This is meant to teach you how to use, not necessarily everything about the language. I think maybe we have fun facts about python as a separate "tip"